src/lib/WalletInfoCollector.ts

```diff
@@ -72,20 +77,33 @@ export default class WalletInfoCollector {
         if (initialAccounts.length > 0) {
             WalletInfoCollector._addAccounts(walletInfo, initialAccounts, undefined);
 
-            if (walletType !== WalletType.LEGACY) {
-                // fetch balances and update again
-                initialAccountsPromise = WalletInfoCollector._getBalances(initialAccounts).then(async (balances) => {
-                    WalletInfoCollector._addAccounts(walletInfo, initialAccounts, balances);
-                    onUpdate(walletInfo, await derivedAccountsPromise);
-                });
-            }
+            // fetch balances and update again
+            // This step could be skipped for WalletType.LEGACY in case there is only one Account to collect info on.
+            // In that case the WalletInfoCollector needs to wait for consensus unnecessarily.
+            // The WalletInfoCollector however has no knowledge about the number of Accounts (yet).
+            initialAccountsPromise = WalletInfoCollector._getBalances(initialAccounts).then(async (balances) => {
+                WalletInfoCollector._addAccounts(walletInfo, initialAccounts, balances);
+                onUpdate(walletInfo, await derivedAccountsPromise);
+            });
+
         }
         onUpdate(walletInfo, await derivedAccountsPromise);
 
-        if (walletType === WalletType.LEGACY) {
+        if (walletType === WalletType.LEGACY && WalletInfoCollector._keyguardClient) {
             // legacy wallets have no derived accounts
             await initialAccountsPromise;
-            return { walletInfo };
+            const address = initialAccounts[0].address;
+            hasHistoryOrBalance = walletInfo.accounts.get(address)!.balance !== undefined &&
```

Comment:
Furthermore, this check should only be run when multiple accounts are checked. Otherwise this check is totally unnecessary.

I thus propose to add a boolean flag to the collector initialization, that tells it if it needs to check this at all.